### PR TITLE
Change TGPPL references to BOSL

### PR DESCRIPTION
**Pull request by steven-ecc**
Updates all TGPPL references to BOSL and makes a few minor cleanups to wording

### CODE REVIEW

**Review by daira — APPROVED** [review]
I have reservations about BOSL wrt license compatibility, but this PR does what it intends to do.

File: Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ edition = "2018"
 description = """
 Fast proof-carrying data implementation with no trusted setup
 """
-license = "TGPPL"
+license = "BOSL"
```
Comment (ebfull) [inline Cargo.toml:14]:
```suggestion
license-file = "LICENSE-BOSL"
```

**Review by ebfull — CHANGES_REQUESTED** [review]

**Comment by gojomo** [comment]
Is "BOSL" mainly a new name for "TGPPL"? (A search for `[bootstrap open source license]` mainly turns up info about the use of the MIT license by the JS/CSS UI project 'Bootstrap'.)

**Comment by steven-ecc** [comment]
> Is "BOSL" mainly a new name for "TGPPL"? (A search for `[bootstrap open source license]` mainly turns up info about the use of the MIT license by the JS/CSS UI project 'Bootstrap'.)

That's correct for the most part Gordon. There's an additional requirement in BOSL that the source code for derived works be published prior to the derived work being made available but that was something we had planned to tweak in the TGPPL anyway.

**Pull request merged**